beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -221,23 +212,78 @@ func (s *Server) GetBlockAttestations(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+	resp := &structs.GetBlockAttestationsResponse{
+		Data:                atts,
+		ExecutionOptimistic: isOptimistic,
+		Finalized:           s.FinalizationFetcher.IsFinalized(ctx, root),
+	}
+	httputil.WriteJson(w, resp)
+}
+
+// GetBlockAttestationsV2 retrieves attestation included in requested block.
+func (s *Server) GetBlockAttestationsV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetBlockAttestationsV2")
+	defer span.End()
+
+	blk, isOptimistic, root := s.blockData(ctx, w, r)
+	consensusAtts := blk.Block().Body().Attestations()
+
+	var atts = make([]interface{}, len(consensusAtts))
+	v := version.String(blk.Block().Version())
+
+	if v == version.String(version.Electra) {
+		for i, att := range consensusAtts {
+			a, ok := att.(*eth.AttestationElectra)
+			if ok {
+				atts[i] = structs.AttElectraFromConsensus(a)
+			} else {
+				httputil.HandleError(w, fmt.Sprintf("unable to convert consensus attestations electra of type %T", att), http.StatusInternalServerError)
+				return
+			}
+		}
+	} else {
+		for i, att := range consensusAtts {
+			a, ok := att.(*eth.Attestation)
+			if ok {
+				atts[i] = structs.AttFromConsensus(a)
+			} else {
+				httputil.HandleError(w, fmt.Sprintf("unable to convert consensus attestations of type %T", att), http.StatusInternalServerError)
+				return
+			}
+		}
+	}
+
+	resp := &structs.GetBlockAttestationsV2Response{
+		Version:             v,
+		ExecutionOptimistic: isOptimistic,
+		Finalized:           s.FinalizationFetcher.IsFinalized(ctx, root),
+		Data:                atts,
+	}
+	httputil.WriteJson(w, resp)
+}
+
+func (s *Server) blockData(ctx context.Context, w http.ResponseWriter, r *http.Request) (interfaces.ReadOnlySignedBeaconBlock, bool, [32]byte) {
+	blockId := r.PathValue("block_id")
+	if blockId == "" {
+		httputil.HandleError(w, "block_id is required in URL params", http.StatusBadRequest)
+		return nil, false, [32]byte{}
+	}
+	blk, err := s.Blocker.Block(ctx, []byte(blockId))
+	if !shared.WriteBlockFetchError(w, blk, err) {
+		return nil, false, [32]byte{}
```

Comment:
doesn't this also need to handle the error? 🤔

Reply:
We don't, `WriteBlockFetchError` handles the error based on different cases

Reply:
there is a code smell here it feels like  that it uses a different pattern

Reply:
I would say the word `Write` suggests that the function will write the error to the output